### DISCUSSION TITLE
Contributions for Google spanner persistor

### DISCUSSION THREAD

**Discussion by dragcp** [ory/keto#590] [Q&A] [answered]
I did read the earlier conversation in [581](https://github.com/ory/keto/discussions/581) 

I understand that Spanner is not a priority. But if I did spend the time to implement this, would you be willing to accept spanner related code into Keto?

If yes, should it just be another struct that implements the Persistor interface with spanner specific code? Or do you recommend a different approach?

**Comment by vinckr** [comment, marked as answer]
I think @zepatrik should be the one to decide this (and give advice on the best approach), but imho we would accept a spanner integration in Keto.
It wont be the primary solution as Patrik pointed out, as that would require open source; but having it as a additional feature sounds good.

**Reply by zepatrik** [reply]
The main problem I see is that we have to maintain it down the road. At least we can write tests and run them for free using https://github.com/GoogleCloudPlatform/cloud-spanner-emulator
I would really like to add that at a later stage, currently we have to add a lot of features going forward. IMO it would be too much work to maintain a spanner persister as long as the internal API is not somewhat stable. Hope you can understand that decision.

**Comment by dragcp** [comment]
Understood. Thank you

**Comment by sp71** [comment]
Since Spanner is not supported directly, you can alternatively use [pgadapter](https://github.com/GoogleCloudPlatform/pgadapter)

> PGAdapter is a proxy that translates the PostgreSQL wire-protocol into the equivalent for Spanner databases [that use the PostgreSQL interface](https://cloud.google.com/spanner/docs/postgresql-interface). It enables you to use standard PostgreSQL drivers and tools with Cloud Spanner and is designed for the [lowest possible latency](https://github.com/GoogleCloudPlatform/pgadapter/blob/postgresql-dialect/benchmarks/latency-comparison/README.md).
